In [20]:
import pandas as pd
import numpy

df = pd.read_excel("/Users/jie/Documents/shopify/ebike/E-Bike_cleaned.xls",index=False)
df_part1=df[['Date', 'Age','Gender','Health','Education','Family_Income','Employment',
              'Address','Travel_distance','Commute','Transportation','Response']]

#########################################
###########Data description##############
df_part1.head()
df_part1.describe(include=['O'])

Age Gender     Health   Education Family_Income Employment  \
count       2234   2221       2227        2220          2178       2223   
unique         5      4          6           7             6          6   
top     35 to 49   Male  Very good  University        $100K+   Employee   
freq         863   1554        891         903           831       1410   

        Address Travel_distance          Commute Transportation  
count      2237            2237             2237           2238  
unique        7               5                6              5  
top     Toronto       5 - 10 km  16 - 29 minutes           Bike  
freq       2145             847              782            785

### What is the distribution of categorical features?
* Each row is unique across the dataset (count=unique=2238).
* Age takes 5 ranges with 38.6% "35 to 49 years old" (top=35 to 49, freq= 863/count=2234).
* Gender variable has 4 possible values, with 70% male (top=male, freq=1554/count=2221).
* Health variable has 6 values, with 40% Very good (top=Very good, freq=891/count=2227).
* Yearly Household Income takes 6 possible values. 100K+ earned by most family (top=100K+)
* Employment has 6 status, most are employee (freq=1410/count=2178).
* Ticket feature has high ratio (22%) of duplicate values (unique=681).

In [ ]:
#divide the dataset into train set and test set, by using the "K-fold" cross-validation



#create dummy variable "Response" by the answer of question 
#"Does your household have access to any of the following types of private motorized vehicles?"
keyword1 = ['yes','car','cars','motorcycle','atv','motor','truck']
keyword2 = ['share','rental','sharing','zip','Car2Go','Cars2Go',"borrow","rent",'carshare']
pat1 = '|'.join(keyword1)
pat2 = '|'.join(keyword2)

df['Response'] = pd.np.where(df['Vehicles'].str.lower().str.contains(pat1), 1, 0)
df['Response'] [df['Vehicles'].str.lower().str.contains(pat2)]=0

#Check the one-way frequency table of "Response"
my_tab = pd.crosstab(index=df['Health'],columns="count") 
my_tab

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn import metrics, cross_validation

df2 = pd.get_dummies(df, columns =['Age','Sex','Health','Education','Family_Income','Employment',
              'Address','Travel_distance'])
df2.head(5)
#logreg=LogisticRegression()
#X, y = df["Age"], df["Response"]


#predicted = cross_validation.cross_val_predict(logreg, df["Age"], df["Response"], cv=10)
#metrics.accuracy_score(y, predicted)



In [ ]:
print metrics.classification_report(y, predicted) 

I have renamed most of the columns for simplify data manipluations. Check reference [1] for details.

## Check missing value pattern

In [ ]:
import missingno as msno
%matplotlib inline

In [ ]:
msno.matrix(df)

At a glance, address and all multiple choice questions regarding with the e-bike use appear to be completed, while the personal information (e.g. household income, education, gender and employment) seems spottier. 


In [ ]:
#Check the randomness of missing value
pd.crosstab(df["Age"].fillna('missing'),df["Family_Income"].fillna('missing'),dropna=False, margins=True)
pd.crosstab(df["Education"].fillna('missing'),df["Family_Income"].fillna('missing'),dropna=False, margins=True)


In [ ]:
msno.heatmap(df)

It seems that survey which is filed with an average travel distance most days of the week is definitely having a complete address; and along with the average commute time variable, travel distance and address will also be provided.

Nullity correlation ranges from -1 (if one variable appears the other definitely does not) to 0 (variables appearing or not appearing have no effect on one another) to 1 (if one variable appears the other definitely also does).

## Suggestions for next survey
1.Age should be treated as continuous variable



#### Reference
##### [1] Table Names

| # |Original Names |   New Column Names   |
| :- |:---:| :-|
| 1|Timestamp | Date|
| 2|1. What age range do you fall in? | Age|
| 3|Sex | Gender|
| 4|How would you describe your level of physical health? | Health|
| 5|What level of education have you reached? | Education|
| 6|What is your household income? | Family_income|
| 7|Which category best describes your employment? | Employment|
| 8|What Toronto district is your primary address located in? | Address|
| 9|On average, what distance do you travel most days of the week? | Travel_distance|
| 10|On average, how long is your commute? (each way) | Commute|
| 11|Which transportation option do you end up using most often? | Transportation|
| 12|Does your household have access to any of the following types of private motorized vehicles? | Vehicles|
| 13|Do you support any of the following statements? | Support_Statements|
| 14|When you use Toronto's Multi-Use Trails do you mostly | Multiuse_Trail|
| 15|Are you aware that the City of Toronto's Multi-Use Paths have a speed limit of 20 km/h? | Trail_speed|
| 16|Have you witnessed a collision or conflict on a trail between | Collision_witness|
| 17|Do you think more should be done to manage trail users who do not respect the 20 km/h speed limit? | Speed_limit|
| 18|Currently, any kind of e-bike may use a multi-use path if they are propelled by pedaling only.  If any e-biker or other type of vehicle is being propelled by motor power, then it is considered a motor vehicle, and may be fined. | multiuselane_propelled|
| 19|When you use Toronto's bicycle lanes do you mostly | bicyclelane_use|
| 20|Currently, any kind of e-bike may use a bicycle lane, provided they are propelled by pedalling only.  If any e-bike is being propelled by motor power, then it is considered a motor vehicle and may be fined.  | bikelane_propelled|
| 21|Sidewalks are for pedestrians.  Cyclists and e-bikers should not ride or drive on sidewalks.  With regards to illegal use of bicycles and e-bikes on sidewalks, should the City. | Sidewalk|
| 22|Toronto Bylaws consider personal mobility devices (such as electric wheel chairs) to be pedestrians.  Historically, there has been little risk as electric wheelchairs available for purchase have only traveled at speeds which are close to walking speed.  More recently, personal mobility devices which resemble e-bikes and may travel at more than triple walking speed have become available.  In your opinion, should the City; | Personal_mobile|
<style>
table {float:left}
</style>